include/dali/core/device_guard.h

```diff
@@ -17,38 +17,26 @@
 
 #include "dali/core/cuda_utils.h"
 #include "dali/core/error_handling.h"
+#include "dali/core/cuda_utils.h"
 
 namespace dali {
-/**
- * Simple RAII device handling:
- * Switch to new device on construction, back to old
- * device on destruction
- */
-class DeviceGuard {
- public:
+
+// /**
+//  * Simple RAII device handling:
+//  * Switch to new device on construction, back to old
+//  * device on destruction
+//  */
+class DLL_PUBLIC DeviceGuard {
+public:
   /// @brief Saves current device id and restores it upon object destruction
-  DeviceGuard() {
-    CUDA_CALL(cudaGetDevice(&original_device_));
-  }
+  DeviceGuard();
+
   /// @brief Saves current device id, sets a new one and switches back
```

Comment:
You should write here that it's supposed to be no-op for -1

Reply:
Done